sql-plugin/src/main/scala/com/nvidia/spark/rapids/GpuWindowExpression.scala

```diff
@@ -362,6 +403,74 @@ object GpuWindowExpression {
     case anythingElse =>
       throw new UnsupportedOperationException(s"Unsupported window frame expression $anythingElse")
   }
+
+  /**
+   * Create a Scalar from boundary value according to order by column type.
+   *
+   * For the timestamp types, only days are supported for window range intervals
+   *
+   * @param orderByType the type of order by column
+   * @param value boundary value
+   * @return Scalar holding boundary value
+   */
+  def createRangeWindowBoundary(orderByType: DType, value: Long): Scalar = {
+    implicit def daysToDuration(days: Long) = Duration(value, DAYS)
+    orderByType match {
+      case DType.INT8 => Scalar.fromByte(value.toByte)
+      case DType.INT16 => Scalar.fromShort(value.toShort)
+      case DType.INT32 => Scalar.fromInt(value.toInt)
+      case DType.INT64 => Scalar.fromLong(value)
+      case DType.TIMESTAMP_DAYS => Scalar.durationFromLong(DType.DURATION_DAYS, value.toDays)
+      case DType.TIMESTAMP_SECONDS => Scalar.durationFromLong(DType.DURATION_SECONDS,
+        value.toSeconds)
+      case DType.TIMESTAMP_MILLISECONDS =>
```

Comment:
I m a bit conflicted by this. Spark only supports TIMESTAMP_DAYS and TIMESTAMP_MICROSECONDS. Why are we including code for all of the others that will never be used?

Reply:
Done. removed other un-supported types